src/nvim/popupmnu.c

```diff
@@ -237,6 +236,10 @@ void pum_display(pumitem_T *array, int size, int selected, bool array_changed,
         pum_height = row - above_row;
       }
     }
+    if (pum_external) {
```

Comment:
this seems quite ad-hoc. vim already carries 1-2 items between pages it seems. I don't think we should do any extra adjustment for `pum_external` only.